cmd/agent/gui/checks.go

```diff
@@ -243,42 +276,81 @@ func listFiles(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(res))
 }
 
-// Helper function which returns the appropriate paths for finding checks/check configs
-func getAllPaths(fileType string) []string {
-	if fileType == "py" {
-		return []string{
-			filepath.Join(common.GetDistPath(), "checks.d"), // Custom checks
-			config.Datadog.GetString("additional_checksd"),  // Custom checks
-			common.PyChecksPath,                             // Integrations-core checks
+// Helper function which returns all the filenames in a check config directory
+func readConfDir(path string) ([]string, error) {
+	var filenames []string
+	entries, err := ioutil.ReadDir(path)
+	if err != nil {
+		return filenames, err
+	}
+
+	for _, entry := range entries {
+		// Some check configs are in nested subdirectories
+		if entry.IsDir() {
+			if filepath.Ext(entry.Name()) != ".d" {
+				continue
+			}
+
+			subEntries, err := ioutil.ReadDir(filepath.Join(path, entry.Name()))
+			if err == nil {
+				checkName := strings.TrimSuffix(entry.Name(), ".d")
+
+				for _, subEntry := range subEntries {
+					if hasRightEnding(subEntry.Name()) && subEntry.Mode().IsRegular() {
+						// Add the check name to the filename so that it's in a format the GUI can easily display
+						filenames = append(filenames, checkName+": "+subEntry.Name())
+					}
+				}
+			}
+			continue
 		}
-	} else if fileType == "yaml" {
-		return []string{
-			config.Datadog.GetString("confd_path"),        // Custom checks
-			filepath.Join(common.GetDistPath(), "conf.d"), // Default check configs
+
+		if hasRightEnding(entry.Name()) && entry.Mode().IsRegular() {
+			filenames = append(filenames, entry.Name())
 		}
 	}
-	return []string{}
+
+	return filenames, nil
 }
 
-// Helper function which returns all the filenames in a directory
-func readDirectory(path string) ([]string, error) {
-	var filenames []string
-	dir, e := os.Open(path)
-	if e != nil {
-		return filenames, e
+// Helper function which checks if a file has a valid extension
+func hasRightEnding(filename string) bool {
+	// Only accept files of the format
+	//	{name}.yaml, {name}.yml
+	//	{name}.yaml.default, {name}.yml.default
+	//	{name}.yaml.example, {name}.yml.example
+
+	ext := filepath.Ext(filename)
+	if ext == ".default" {
+		ext = filepath.Ext(strings.TrimSuffix(filename, ".default"))
+	} else if ext == ".example" {
+		ext = filepath.Ext(strings.TrimSuffix(filename, ".example"))
 	}
-	defer dir.Close()
 
-	files, e := dir.Readdir(-1)
-	if e != nil {
-		return filenames, e
-	}
+	return ext == ".yaml" || ext == ".yml"
+}
 
-	for _, file := range files {
-		if file.Mode().IsRegular() {
-			filenames = append(filenames, file.Name())
-		}
+// Helper function which returns the path where a file should be stored
+func getPathToFile(fileName string) string {
+	// If a file is in a subdir, it's name will be of the form `{check name}: {file name}`
+	// We want to change this to '{check name}.d/{file name}'
```

Comment:
why don't we display `{check name}.d: {file name}` ? Or event the full path `{check_name.d}/{filename}` ?

Reply:
Sure, let's do that! I originally didn't want to because I thought it might be more confusing for a GUI user who doesn't know about the ".d" directories, but since using the full path on the back end will remove the need for those two extra helper functions it's probably worth it to just use tgat and not bother with changing the formatting